provider/provider_yaml_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
+
+//go:build !go && !nodejs && !python && !dotnet
+// +build !go,!nodejs,!python,!dotnet
+
+package provider
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/pulumi/providertest"
+)
+
+func TestBucket(t *testing.T) {
+	test(t, filepath.Join("test-programs", "bucket"))
+}
+
```

Comment:
I'm curious why you switched to making a separate test function for each program rather than just iterating the test directory and generating one per program.

Reply:
I believe I'm taking style guidance from @danielrbradley on this; both are idiomatic Go as far as I can tell. This style is even easier to point-and-click running a test in VS Code.